src/configureWorkspace/configurePython.ts

```diff
@@ -3,68 +3,235 @@
+        folder: context.folder,
+        platform: 'python',
+        actionContext: context,
+        dockerfile: dockerfile,
+        generateComposeTask: generateComposeFiles,
+        ports: ports
+    }
+
+    const pyOptions: PythonScaffoldingOptions = {
+        projectType: projectType,
+        target: target
+    }
+
+    await dockerDebugScaffoldingProvider.initializePythonForDebugging(scaffoldContext, pyOptions);
+}
+
+export async function promptForLaunchFile(projectType?: PythonProjectType) : Promise<PythonTarget> {
+    const launchFilePrompt = defaultLaunchFile.get(projectType);
+
+    const opt: vscode.InputBoxOptions = {
+        placeHolder: launchFilePrompt.defaultFile,
+        prompt: launchFilePrompt.prompt,
+        value: launchFilePrompt.defaultFile,
+        validateInput: (value: string): string | undefined => { return value && value.trim().length > 0 ? undefined : 'Enter a valid Python file path/module.' }
+    };
+
+    const file = await ext.ui.showInputBox(opt);
+
+    // If the input has the .py extension, then assume it is a file.
+    if (path.extname(file).toLocaleUpperCase() === PythonFileExtension.toLocaleUpperCase()) {
+        return { file: file.replace(/\\/g, '/') };
+    } else {
+        return { module: file};
+    }
+}
+
+export async function scaffoldPython(context: ScaffolderContext): Promise<ScaffoldFile[]> {
+    const properties: TelemetryProperties & ConfigureTelemetryProperties = context.telemetry.properties;
+    const serviceName = context.folder.name;
+    const rootFolderPath: string = context.rootFolder;
+    const outputFolder = context.outputFolder ?? rootFolderPath;
+
+    const generateComposeFiles = await context.captureStep('compose', quickPickGenerateComposeFiles)();
+    const projectType = getPythonProjectType(context.platform);
+
+    const defaultPort = PythonDefaultPorts.get(projectType);
+    let ports = [];
+
+    if (defaultPort) {
+        ports = await context.promptForPorts([ defaultPort ]);
+    }
+
+    const launchFile = await context.captureStep('pythonFile', promptForLaunchFile)(projectType);
+    const dockerFileContents = genDockerFile(serviceName, launchFile, projectType, ports);
+
+    const files: ScaffoldFile[] = [
+        { fileName: 'Dockerfile', contents: dockerFileContents, open: true },
+        { fileName: '.dockerignore', contents: genCommonDockerIgnoreFile(context.platform) }
+    ];
+
+    const requirementsFileExists = await fse.pathExists(path.join(outputFolder, 'requirements.txt'));
+
+    if (!requirementsFileExists) {
+        files.push({ fileName: 'requirements.txt', contents: genRequirementsFile(projectType) });
+    }
+
+    if (generateComposeFiles) {
+        properties.orchestration = 'docker-compose';
+
+        const dockerComposeFile = genDockerCompose(serviceName, ports);
+        const dockerComposeDebugFile = await genDockerComposeDebug(serviceName, projectType, ports, launchFile);
+
+        files.push(
```

Comment:
Do you have special logic to create the file with *.(n) if the file exists?

Reply:
No, that wouldn't be part of the PR. I'm not sure 100% if we need it for Python but we can discuss it at some point.

Reply:
I believe we want the same compose behavior across all platform. it is possible, use might have had another service added to the existing compose and we don't want to overwrite it. In any case, I am adding this functionality to my PR and you could make use of it at later point.

Reply:
Cool, thanks for the heads up, @ravipal. I'd definitely re-use whatever you have at that point.